Fix a few defects highlighted by the Coverty scan

dalg24 opened the pull request.

Commit:
Commenting out unreachable code

Commit:
Check return value code when calling std::remove from cstdio

Commit:
Add exception handling to cpp examples

Getting rid of uncaught exception issue pointed out by Coverty scan

Rombur commented on cpp/source/deal.II/supercapacitor.templates.h:
```diff
@@ -228,8 +228,12 @@ void SuperCapacitor<dim>::evolve_one_time_step_constant_load(
+  std::ignore = load;
   throw std::runtime_error("This function is not implemented.");
-  evolve_one_time_step(time_step, ConstantLoad, true);
+  // Commenting out the next line to get rid of structurally dead code issue
```
Just remove the line of code. There is no reason to keep it.

Rombur reviewed (approved):
Looks good. Just remove the dead code and merge yourself.

Commit:
Improve coverage

Check that evolve_one_time_step_constant_load() function that has not been
implemented yet for the supercapacitor is throwing a runtime exception.
